debezium-embedded/src/main/java/io/debezium/embedded/ConnectorEngine.java

```diff
@@ -0,0 +1,1548 @@
+
+        // Create the worker config, adding extra fields that are required for validation of a worker config
+        // but that are not used within the embedded engine (since the source records are never serialized) ...
+        Map<String, String> embeddedConfig = config.asMap(ALL_FIELDS);
+        embeddedConfig.put(WorkerConfig.KEY_CONVERTER_CLASS_CONFIG, JsonConverter.class.getName());
+        embeddedConfig.put(WorkerConfig.VALUE_CONVERTER_CLASS_CONFIG, JsonConverter.class.getName());
+        workerConfig = new EmbeddedWorkerConfig(embeddedConfig);
+
+        // Create the offset store implementation ...
+        offsetStore = config.getInstance(OFFSET_STORAGE, OffsetBackingStore.class, () -> this.classLoader);
+        logger.debug("Create connector engine '{}' with configuration: {}", name, config.withMaskedPasswords());
+    }
+
+    /**
+     * Get the name of this engine.
+     * 
+     * @return the engine name; never null
+     */
+    protected String name() {
+        return this.config.getString(NAME);
+    }
+
+    /**
+     * Start this engine's resources and any deployed connectors. Once started, any additional connectors will be started
+     * immediately when they are {@link #addConnector(Configuration, ConnectorCallback, ClassLoader) deployed}.
+     * 
+     * @see #stopAllConnectors(Runnable)
+     * @see #close()
+     */
+    public synchronized void start() {
+        checkStatus();
+        if (running.compareAndSet(false, true)) {
+            logger.debug("Starting connector engine '{}'", name, offsetStore);
+            try {
+                // Instantiate and initialize the offset store ...
+                offsetStore = config.getInstance(OFFSET_STORAGE, OffsetBackingStore.class, () -> this.classLoader);
+                offsetStore.configure(workerConfig);
+                offsetStore.start();
+
+                // Set up the offset commit policy ...
+                long offsetPeriodMs = config.getLong(OFFSET_FLUSH_INTERVAL_MS);
+                offsetCommitPolicy = OffsetCommitPolicy.periodic(offsetPeriodMs, TimeUnit.MILLISECONDS);
+
+                // Start any existing connector workers ...
+                workers.forEach((connectorName, worker) -> {
+                    logger.debug("Starting connector '{}' in engine '{}'", connectorName, name());
+                    worker.start();
+                });
+            } catch (Throwable t) {
+                running.set(false);
+            }
+        }
+    }
+
+    /**
+     * Wrap the supplied offset storage reader with a delegate implementation that also logs the partitions and offsets that are
+     * read.
+     * 
+     * @param reader the actual offset reader; may not be null
+     * @param logger the logger to which trace messages should be written; may not be null
+     * @return the instrumented reader implementation; never null
+     */
+    protected static OffsetStorageReader instrument(OffsetStorageReader reader, Logger logger) {
```

Comment:
maybe call this method `log` or `debug` or something. `instruments` seems very generic to me and my first thought is Dynamic Proxies :)